realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -458,6 +458,11 @@ public void close() {
      * Closes the Realm instances and all its resources without checking the {@link RealmCache}.
      */
     void doClose() {
+        if (asyncTaskExecutor.hasPendingTransactions() || !handlerController.asyncTransactionCallbacks.isEmpty()) {
```

Comment:
Since it is possible to deduce if we have pending transactions or callback (or both), we could log more precisely. It might help our user.